netty/src/main/java/io/grpc/transport/netty/NettyClientStream.java

```diff
@@ -132,14 +132,8 @@ protected void sendFrame(WritableBuffer frame, boolean endOfStream, boolean flus
       onSendingBytes(numBytes);
       writeQueue.enqueue(
           new SendGrpcFrameCommand(this, bytebuf, endOfStream),
-          channel.newPromise().addListener(new ChannelFutureListener() {
-            @Override
-            public void operationComplete(ChannelFuture future) throws Exception {
-              // Remove the bytes from outbound flow control, optionally notifying
-              // the client that they can send more bytes.
-              onSentBytes(numBytes);
-            }
-          }), flush);
+          channel.voidPromise(),
```

Comment:
We listen to the promise for almost every command we dispatch through the handler so it's cleaner to opt in to void than to opt out of it.